src/components/App/App.actions.js

```diff
@@ -46,3 +48,55 @@ export function updateUserData(userData) {
     userData
   };
 }
+
+export function setUnloggedUserLocation(location) {
+  return {
+    type: SET_UNLOGGED_USER_LOCATION,
+    location
+  };
+}
+
+export function updateLoggedUserLocation() {
+  return async (dispatch, getState) => {
+    const {
+      app: { userData }
+    } = getState();
+    if (!userData) return;
+    try {
+      const { id, location } = userData;
+      const APIGetAndUpdateLocation = async () => {
+        const { location: userLocation } = await API.updateUser({
+          id: id,
+          location: {}
+        });
+        return userLocation;
+      };
+
+      if (location) return;
+      const userLocation = await APIGetAndUpdateLocation();
+      if (userLocation)
+        return dispatch(
```

Comment:
This looks strange, a function should not return a dispatcher. Although it works, it would be more readable if first dispatch and then return.